model_analyzer/triton/model/model_config.py

```diff
@@ -0,0 +1,144 @@
+# Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from .model_config_utils import parse_model_config
+from model_analyzer.model_analyzer_exceptions \
+    import TritonModelAnalyzerException
+
+
+class ModelConfig:
+    """
+    A class that encapsulates
+    all the metadata about
+    a model.
+    """
+
+    def __init__(self, model_path, instance_group=None, dynamic_batching=None):
+        """
+        Parameters
+        -------
+        model_path : str
+            The full path to this model directory
+        instance_group : dict
+            instance_group with KIND and count to initialize
+            this model config
+        dynamic_batching : dict
+            to request dynamic batching for this model
+        """
+
+        self._model_path = model_path
+        self._config = parse_model_config(model_path)
+        if instance_group:
+            self.set_instance_group(instance_group)
+        if dynamic_batching:
+            self.set_dynamic_batching(dynamic_batching)
+
+    def name(self):
+        """
+        Returns
+        -------
+        str
+            The name of this model
+        """
+
+        if 'name' not in self._config:
+            raise TritonModelAnalyzerException(
+                f"Model's name not found in ModelConfig at {self._model_path}")
+        return self._config['name']
+
+    def platform(self):
+        """
+        Returns
+        -------
+        str
+            The platform name of this model
```

Comment:
What is platform? Is it cpu/gpu?

Reply:
I believe this is the backend name. (TF, ONNX, libtorch) etc

Reply:
I see.